wandb/integration/yolov8/yolov8.py

```diff
@@ -184,6 +196,41 @@ def teardown(self, _trainer: BaseTrainer) -> None:
         self.run.finish()
         self.run = None
 
+    def on_val_start(
+        self,
+        validator: BaseValidator,
+    ) -> None:
+        """On validation start we create a wandb table to store the sample predictions
+        and patch the validator's `plot_predictions` method to plot the predictions to the wandb table .
+        """
+        class_names = sorted(validator.names.values())
+        if isinstance(validator, DetectionValidator):
+            self.validation_table = wandb.Table(
+                columns=["Batch", "Ground Truth", "Prediction", *class_names]
+            )
+            self.og_plot_predictions = validator.plot_predictions
+            validator.plot_predictions = partial(
+                plot_detection_predictions, self, validator
+            )
+
+        if isinstance(validator, SegmentationValidator):
+            self.validation_table = wandb.Table(
+                columns=["Batch", "Ground Truth", "Prediction", *class_names]
+            )
+            self.og_plot_predictions = validator.plot_predictions
+            validator.plot_predictions = partial(
```

Comment:
is it ok that we just overwrite the original plotting logic? maybe it makes sense to append?
I'm not sure how the integrations work, but on the face of it seems like it would break users that expect the original behavior of the validator?